### PR TITLE
MSEXLSTa 1.2

### PR DESCRIPTION
**Pull request by spark404**
Recently needed a protocol disector for CITP. Thanks for making this.

I've added a decoder for the LSTa 1.2 message and a small fix to show the name in the PLoc name field.

### CODE REVIEW

File: citp.lua [hunk 1]
```diff
@@ -603,79 +603,161 @@ function citp_proto.dissector(buffer,pinfo,tree)
...
+            str = "None, "
+          end
+
+          str = string.sub(str,1,-3)
+
+          LSta[i]:add(buffer(start,count), "Layer Status: ".."("..current_stat..") "..str)
         end
 
-        str = string.sub(str,1,-3)
+        if version == "1.2" then
```
Comment (hossimo) [inline citp.lua]: Convert version check to if/elseif pattern

File: citp.lua [hunk 2]
```diff
@@ -603,79 +603,161 @@ function citp_proto.dissector(buffer,pinfo,tree)
...
+          end
+          if bit.band(current_stat,00000010) > 0 then -- 1.2 Only
+            str = str .. "MediaPlaybackRandom, "
+          end
+          if bit.band(current_stat,00000020) > 0 then -- 1.2 Only
+            str = str .. "MediaPaused, "
+          end
+          if current_stat == "00000000" then
+            str = "None, "
+          end
```
Comment (hossimo) [inline citp.lua]: Convert version check to if/elseif pattern

**Review by hossimo — CHANGES_REQUESTED** [review]
Thanks for the PR. The changes took good. the only thing I noticed is two instances of ifs that could be treated like elseif.

I haven't looked at this code in some time so there are probably a number of instances like this that I should probably fix.

File: citp.lua (hunk 2 above)
Comment (spark404) [reply citp.lua]: I think you did the right thing here with the multiple if constructs. This is a flag field and multiple flags could be set.

**Comment by spark404** [comment]
> Thanks for the PR. The changes took good. the only thing I noticed is two instances of ifs that could be treated like elseif.
> 
> I haven't looked at this code in some time so there are probably a number of instances like this that I should probably fix.

There is always something to fix, yet it does its job 👍

File: citp.lua (hunk 2 above)
Comment (hossimo) [reply citp.lua]: ahh yes you are correct, i forgot that was the bit.band was the construct bitwise ands.

**Comment by hossimo** [comment]
Awesome, Thanks! 👍

**Pull request merged**